TombLib/TombLib/Catalogs/TEN Node Catalogs/Environment.lua

```diff
@@ -517,35 +497,56 @@ LevelFuncs.Engine.Node.ChangeHorizonRotationOverTimespan = function(type, relati
 	end
 end
 
--- !Name "Change transparency of horizon over specified time"
+-- !Name "Change transparency of horizon over time"
 -- !Section "Environment"
 -- !Description "Gradually change transparency over specified timespan."
 -- !Arguments "NewLine, Enumeration, 25, [ Horizon 1 | Horizon 2 ], {0}, Horizon"
--- !Arguments "Numerical, 15, [ 0 | 1 | 2 | 0.1 | 1 ], {1} Transparency" 
--- !Arguments "Numerical, [ 0.1 | 65535 | 2 | 0.1 | 1 ], {1}, 25, Time (in seconds)"
+-- !Arguments "Numerical, 20, [ 0 | 1 | 2 | 0.1 | 1 ], {1} Transparency" 
+-- !Arguments "Numerical, [ 0.1 | 65535 | 2 | 0.1 | 1 ], {1}, 20, Time (in seconds)"
 -- !Arguments "Boolean, 35, Relative coordinates"
- 
-
 LevelFuncs.Engine.Node.ChangeHorizonTransparencyOverTimespan = function(type, newTransparency, time, relative)
 	
 	if (type == 0) then
 		if (relative) then
 			newTransparency = TEN.Flow.GetCurrentLevel().horizon1.transparency + newTransparency
 		end
-		do LevelFuncs.Engine.Node.ConstructWeatherTimedData(8, 4, newTransparency, time, smooth) end	
+		do LevelFuncs.Engine.Node.ConstructWeatherTimedData(8, 4, newTransparency, time, true) end	
 	elseif (type == 1) then
 		if (relative) then
 			newTransparency = TEN.Flow.GetCurrentLevel().horizon2.transparency + newTransparency
 		end
-		do LevelFuncs.Engine.Node.ConstructWeatherTimedData(11, 4, newTransparency, time, smooth) end
+		do LevelFuncs.Engine.Node.ConstructWeatherTimedData(11, 4, newTransparency, time, true) end
+	end
+	
+end
+
+-- !Name "Fade in a new horizon over time"
+-- !Section "Environment"
+-- !Description "Gradually fade in a horizon in the other slot over specified timespan."
+-- !Arguments "Enumeration, 25, [ Horizon 1 | Horizon 2 ], {0}, Fade out which horizon"
+-- !Arguments "NewLine, WadSlots, 75, {TEN.Objects.ObjID.HORIZON}, Fade in which horizon"
+-- !Arguments "Numerical, 25 [ 0.1 | 65535 | 2 | 0.1 | 1 ], {1}, Time (in seconds)"
+LevelFuncs.Engine.Node.FadeHorizonTransparencyOverTimespan = function(type, slot, time)
+	
+	if (type == 0) then
```

Comment:
I recommend to automatically manage type of the fade (i.e. which horizon to show and which to hide) by comparing alpha values of two horizons. One with smaller alpha value must be faded in. This way you can ping-pong horizons without caring about whether it's `horizon1` or `horizon2`.